### PR TITLE
chore: sync codebase from expo sdk for 3.0.0

### PR DESCRIPTION
**Pull request by michaellee8**

### CODE REVIEW

File: src/Web3Auth.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      init: initParams,
+      params: {
+        ...params,
+        ...(!params.redirectUrl && { redirectUrl: redirectUrl }),
+      },
+    };
+
+    console.log(`[Web3Auth] params passed to Web3Auth: ${mergedParams}`);
+
+    const hash = Base64.encodeURI(JSON.stringify(mergedParams));
```
Comment (chaitanyapotti) [inline src/Web3Auth.ts:42]: Need to use base64url

File: src/Web3Auth.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    const url = new URL(this.initParams.sdkUrl);
+    url.pathname = url.pathname + `${path}`;
+    url.hash = hash;
+
+    console.log(
```
Comment (chaitanyapotti) [inline src/Web3Auth.ts:48]: Use loglevel instead

File: openlogin-expo-sdk-example-expo/App.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import Web3Auth, {
+  LOGIN_PROVIDER,
+  OPENLOGIN_NETWORK,
+  SdkInitParams,
+} from "web3auth-react-native-sdk";
+import Constants, { AppOwnership } from "expo-constants";
+import * as Linking from "expo-linking";
+import * as WebBrowser from "expo-web-browser";
+
+const scheme = "openloginexposdkexampleexpo";
```
Comment (chaitanyapotti) [inline openlogin-expo-sdk-example-expo/App.tsx:13]: rename to web3auth pls

**Review by michaellee8 — COMMENTED** [review]
lgtm

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**